File: Backend/Models/Doctor.cs
using System.ComponentModel.DataAnnotations;

namespace QuanLyBenhVien.API.Models;

public class Doctor
{
    [Key]
    public int DoctorID { get; set; }
    public int? UserID { get; set; }
    public int? DepartmentID { get; set; }
    public string? Specialty { get; set; }
    public string? WorkingSchedule { get; set; }
    public string? Experience { get; set; }
    public string? Education { get; set; }
    public decimal? ConsultationFee { get; set; }
    public string? Status { get; set; }
    public int? RoomID { get; set; }

    // Navigation properties
    public virtual User? User { get; set; }
    public virtual Department? Department { get; set; }
    public virtual Room? Room { get; set; }
    public virtual ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
    public virtual ICollection<MedicalRecord> MedicalRecords { get; set; } = new List<MedicalRecord>();
    public virtual ICollection<DoctorSchedule> Schedules { get; set; } = new List<DoctorSchedule>();
}

File: SpringBoot/src/main/java/com/hospital/qlbv/entity/Doctor.java
package com.hospital.qlbv.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;

@Entity
@Table(name = "doctors")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Doctor {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer doctorId;
    
    @Column(name = "user_id")
    private Integer userId;
    
    @Column(name = "department_id")
    private Integer departmentId;
    
    private String specialty;
    
    @Column(name = "working_schedule")
    private String workingSchedule;
    
    private String experience;
    private String education;
    
    @Column(name = "consultation_fee")
    private BigDecimal consultationFee;
    
    private String status;
    
    @Column(name = "room_id")
    private Integer roomId;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    @JsonIgnoreProperties({"passwordHash", "userRoles", "email", "phone", "fullName"})
    private User user;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "department_id", insertable = false, updatable = false)
    @JsonIgnoreProperties({"doctors", "rooms"})
    private Department department;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "room_id", insertable = false, updatable = false)
    @JsonIgnoreProperties({"doctors", "department"})
    private Room room;
}
